src/main/java/mate/jdbc/dao/ManufacturerDaoImp.java

```diff
@@ -0,0 +1,120 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+import mate.jdbc.util.DataProcessingException;
+
+@Dao
+public class ManufacturerDaoImp implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertManufacturer = "INSERT INTO manufacturers(name, country) VALUES(?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createManufacturers = connection
+                        .prepareStatement(insertManufacturer, Statement.RETURN_GENERATED_KEYS)) {
+            createManufacturers.setString(1, manufacturer.getName());
+            createManufacturers.setString(2, manufacturer.getCountry());
+            createManufacturers.executeUpdate();
+            ResultSet generatedKeys = createManufacturers.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert manufacturer: " + manufacturer
+                    + " to DB", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        String getStatement = "SELECT * FROM manufacturers WHERE id = ? "
+                + "AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturer = connection
+                        .prepareStatement(getStatement)) {
+            getManufacturer.setLong(1, id);
+            ResultSet resultSet = getManufacturer.executeQuery();
+            if (!resultSet.next()) {
+                throw new DataProcessingException("Can't get manufacturer. Wrong id: "
+                        + id, new Throwable());
+            }
+            Manufacturer manufacturer = generateManufacturer(resultSet);
+            return Optional.of(manufacturer);

```

Comment:
Do not throw new DataProcessingException in case of the wrong id - we are using Optional here
```suggestion
            if (resultSet.next()) {
                Manufacturer manufacturer = generateManufacturer(resultSet);
                return Optional.of(manufacturer);
            }
            return Optional.empty();
```